### DISCUSSION TITLE
Procedure specific custom request headers

### DISCUSSION THREAD

**Discussion by skovhus** [trpc/trpc#2017] [Q&A]
From the documentation and code example it seems like Client side HTTP headers can only be configured in the tRPC client and not call site when doing a query or mutation. 

But there are use cases where you want to pass in specific headers for a single procedure. What is the recommended way here? Is that something we lack support for? 

Example use case: Elevated privileges that would update the authorization header for a single procedure.

**Comment by KATT** [comment]
The reason why this isn't exposed is that batching makes it a bit tricky to do so and configuring that with the links aren't that straightforward.

Could you give a few ideas of how you would see that working together with the batching and split links? Would it be enough that you could get context data passed through to the headers callback?

**Reply by KATT** [reply]
Move to #2018

**Reply by skovhus** [reply]
I didn’t think about the batching feature here, but I see that as a problem. In my use case elevated auth headers should only be passed on for some procedures. So simply sending the same headers out to the entire batch wouldn’t work. 

Not sure how this is abstracted away, but would it be possible to not batch for procedure calls that specified a custom header?

**Reply by skovhus** [reply]
Should we move the discussion to the other issue?
